src/borg/item.pyx

```diff
@@ -261,7 +262,7 @@ cdef class Item(PropDict):
 
     # properties statically defined, so that IDEs can know their names:
 
-    path = PropDictProperty(str, 'surrogate-escaped str')
+    path = PropDictProperty(str, 'surrogate-escaped str', encode=assert_sanitized_path, decode=to_sanitized_path)
```

Comment:
so the decode here is only to support borg1 archives, while the encode better stays forever?

Reply:
In borg2 archives, path will be sanitize but we'll still need a check for malicious paths (i.e.. absolute paths and '..'.

Reply:
In borg2 archives, path will be sanitize but we'll still need a check for malicious paths (i.e.. absolute paths and '..'. Adding comment to that regard …